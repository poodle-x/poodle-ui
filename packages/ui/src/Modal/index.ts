import React from "react";
import { Modal, ModalProps } from "./Modal";
import { StandardThemeConfig } from "../theme";

export * from "./Modal";

export default Modal;

export type ModalStyleKeys =
	| "root"
	| "overlay"
	| "inner"
	| "container"
	| "modal";

export interface ModalThemeConfig
	extends StandardThemeConfig<ModalStyleKeys, ModalProps> {}

export interface ModalContextValue {
	autoIdLabelledby?: string;
	autoIdDescribedby?: string;
}

export const ModalContext = React.createContext<ModalContextValue>({});
