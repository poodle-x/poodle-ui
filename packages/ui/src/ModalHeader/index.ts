import { ModalHeader, ModalHeaderProps } from "./ModalHeader";
import { StandardThemeConfig } from "../theme";

export * from "./ModalHeader";

export default ModalHeader;

export type ModalHeaderStyleKeys = "root" | "content" | "close" | "icon";

export interface ModalHeaderThemeConfig
	extends StandardThemeConfig<ModalHeaderStyleKeys, ModalHeaderProps> {}
