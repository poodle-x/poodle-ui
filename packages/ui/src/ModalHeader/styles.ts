import { createCSSSystemStandard } from "../theme";
import { CSSObjectSystem, CSSSystem } from "../styled";
import { ModalHeaderProps } from "./ModalHeader";
import { ModalHeaderStyleKeys } from "./index";

export function Root(props: ModalHeaderProps): CSSSystem {
	const { theme } = props;

	return createCSSSystemStandard<ModalHeaderStyleKeys>({
		key: "root",
		config: theme?.ModalHeader,
		props,
		base: [
			{
				pt: "scale-4",
				px: "scale-6",
				display: "flex",
				alignItems: "center",
			},
		],
	});
}

export function Content(props: ModalHeaderProps): CSSSystem {
	const { theme } = props;

	return createCSSSystemStandard<ModalHeaderStyleKeys>({
		key: "content",
		config: theme?.ModalHeader,
		props,
		base: [
			{
				display: "flex",
				alignItems: "center",
				my: 0,
			},
		],
	});
}

export function Close(props: ModalHeaderProps): CSSSystem {
	const { theme } = props;

	return createCSSSystemStandard<ModalHeaderStyleKeys>({
		key: "close",
		config: theme?.ModalHeader,
		props,
		base: [
			{
				ml: "auto",
			},
		],
	});
}

export function Icon(props: ModalHeaderProps): CSSSystem {
	const { theme, colorStyle } = props;

	let colorStyles: CSSObjectSystem = {};

	switch (colorStyle) {
		case "primary": {
			colorStyles = {
				color: "primary.base",
			};
			break;
		}
		case "negative": {
			colorStyles = {
				color: "negative.base",
			};
			break;
		}

		case "positive": {
			colorStyles = {
				color: "positive.base",
			};
			break;
		}

		case "warn": {
			colorStyles = {
				color: "warn.base",
			};
			break;
		}
	}

	return createCSSSystemStandard<ModalHeaderStyleKeys>({
		key: "icon",
		config: theme?.ModalHeader,
		props,
		base: [
			{
				height: "2rem",
				width: "2rem",
				fontSize: "2rem",
				mr: "scale-2",
			},
			colorStyles,
		],
	});
}
