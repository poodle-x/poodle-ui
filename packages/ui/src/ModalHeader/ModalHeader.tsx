import React from "react";
import Box, { BoxProps } from "../Box";
import { IconButton } from "../IconButton";
import Clear from "../icons/Clear";
import { ModalContext } from "../Modal";
import { getCSSSystemBoxProps } from "../styled/system";
import { StandardComponentProps, ThemeConfig } from "../theme";
import useDefaultProps from "../hooks/useDefaultProps/useDefaultProps";
import * as styles from "./styles";

export interface LocalModalHeaderProps {
	children?: React.ReactNode;
	/**
	 * Add close icon button
	 */
	withClose?: boolean;
	/**
	 * Custom props for close icon button
	 */
	closeProps?: BoxProps;
	/**
	 * Add icon to the header
	 */
	icon?: React.ReactNode;
	/**
	 * Custom props for icon element
	 */
	iconProps?: BoxProps;
	/**
	 * Custom props for content element
	 */
	contentProps?: BoxProps;
}

export interface ModalHeaderProps
	extends BoxProps,
		StandardComponentProps,
		LocalModalHeaderProps {}

function getDefaultProps(theme?: ThemeConfig) {
	return theme?.ModalHeader?.defaultProps;
}

export const ModalHeader: React.ForwardRefExoticComponent<
	React.PropsWithoutRef<ModalHeaderProps> & React.RefAttributes<HTMLElement>
> = React.forwardRef<HTMLElement, ModalHeaderProps>((_props, ref) => {
	const { props, isLocalTheme } = useDefaultProps<ModalHeaderProps>(_props, {
		themeDefaultProps: getDefaultProps,
	});

	const {
		children,
		className,
		closeProps,
		withClose,
		icon,
		contentProps,
		iconProps,
		...otherProps
	} = props;

	const { autoIdLabelledby } = React.useContext(ModalContext);

	return (
		<Box
			textStyle="modalTitle"
			id={autoIdLabelledby}
			{...otherProps}
			{...getCSSSystemBoxProps({
				isRoot: true,
				isLocalTheme,
				componentProps: props,
				fnCSSSystem: styles.Root,
				baseClassName: ["poodle-modal-header"],
			})}
			ref={ref}
		>
			<Box
				as="h2"
				{...getCSSSystemBoxProps({
					isLocalTheme,
					componentProps: props,
					partProps: contentProps,
					fnCSSSystem: styles.Content,
					baseClassName: ["poodle-modal-header__content"],
				})}
			>
				{icon && (
					<Box
						{...getCSSSystemBoxProps({
							isLocalTheme,
							componentProps: props,
							partProps: iconProps,
							fnCSSSystem: styles.Icon,
							baseClassName: ["poodle-modal-header__icon"],
						})}
					>
						{icon}
					</Box>
				)}
				{children}
			</Box>
			{withClose && (
				<IconButton
					aria-label="Close"
					{...getCSSSystemBoxProps({
						isLocalTheme,
						componentProps: props,
						partProps: closeProps,
						fnCSSSystem: styles.Close,
						baseClassName: ["poodle-modal-header__close"],
					})}
				>
					<Clear />
				</IconButton>
			)}
		</Box>
	);
});

ModalHeader.displayName = "PoodleModalHeader";
