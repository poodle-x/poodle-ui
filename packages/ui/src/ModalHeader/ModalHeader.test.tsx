import React from "react";
import { ModalContext } from "../Modal";
import { ThemeConfig } from "../theme";
import { render } from "../utils/test";
import ModalHeader from "./";

describe("<ModalHeader />", () => {
	it("should render ModalHeader correctly", () => {
		const { container } = render(<ModalHeader>ModalHeader</ModalHeader>);
		expect(container.firstChild).toMatchSnapshot();
	});

	it("should render ModalHeader with auto id correctly", () => {
		const { container } = render(
			<ModalContext.Provider
				value={{
					autoIdLabelledby: "id",
				}}
			>
				<ModalHeader>ModalHeader</ModalHeader>
			</ModalContext.Provider>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it("should render ModalHeader with icon correctly", () => {
		const { container } = render(
			<ModalHeader icon={<i>Icon</i>}>ModalHeader</ModalHeader>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it("should render ModalHeader with close correctly", () => {
		const { container } = render(
			<ModalHeader withClose={true}>ModalHeader</ModalHeader>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it("ref should work", () => {
		const ref = React.createRef<HTMLDivElement>();

		render(<ModalHeader ref={ref}>ModalHeader</ModalHeader>);
		expect(ref.current).toMatchSnapshot();
	});

	it("should render Box props correctly", () => {
		const { container } = render(
			<ModalHeader
				id="id"
				mt="20px"
				closeProps={{ pl: "20px" }}
				iconProps={{ pl: "20px" }}
				withClose={true}
				icon={<i>Icon</i>}
			>
				ModalHeader
			</ModalHeader>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	describe("theming", () => {
		const themeTest: ThemeConfig = {
			mode: "mode",
			ModalHeader: {
				styles: {
					root: {
						mt: 10,
						letterSpacing: 250,
					},
					icon: {
						mt: 11,
						letterSpacing: 251,
					},
					close: {
						mt: 12,
						letterSpacing: 252,
					},
					content: {
						mt: 13,
						letterSpacing: 253,
					},
				},
				variants: {
					custom: {
						root: {
							top: 0,
							textIndent: 1,
						},
						icon: {
							top: 1,
							textIndent: 2,
						},
						close: {
							top: 2,
							textIndent: 3,
						},
						content: {
							top: 3,
							textIndent: 4,
						},
					},
				},
			},
		};

		it("should extend styles in global", () => {
			const { container } = render(
				<ModalHeader variant="custom" withClose={true} icon={<i>Icon</i>}>
					ModalHeader
				</ModalHeader>,
				{
					theme: themeTest,
				}
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should extend styles in local", () => {
			const { container } = render(
				<ModalHeader
					variant="custom"
					themeExtend={themeTest}
					withClose={true}
					icon={<i>Icon</i>}
				>
					ModalHeader
				</ModalHeader>
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should render with default props correctly", () => {
			const { container } = render(
				<ModalHeader withClose={true} icon={<i>Icon</i>}>
					ModalHeader
				</ModalHeader>,
				{
					theme: {
						ModalHeader: {
							defaultProps: {
								id: "id",
								className: "hello",
							},
						},
					},
				}
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should override styles", () => {
			const { container } = render(
				<ModalHeader withClose={true} icon={<i>Icon</i>}>
					ModalHeader
				</ModalHeader>,
				{
					theme: {
						ModalHeader: {
							overrides: {
								root: {
									mt: 50,
								},
								icon: {
									mt: 51,
								},
								close: {
									mt: 52,
								},
								content: {
									mt: 53,
								},
							},
						},
					},
				}
			);
			expect(container.firstChild).toMatchSnapshot();
		});
	});
});
