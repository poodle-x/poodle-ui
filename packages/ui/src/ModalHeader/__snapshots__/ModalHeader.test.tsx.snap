// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalHeader /> ref should work 1`] = `
.c0 {
  padding-top: 16px;
  padding-left: 24px;
  padding-right: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-style: normal;
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 133.33%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 0;
  margin-bottom: 0;
}

<div
  class="poodle-box poodle-modal-header c0"
>
  <h2
    class="poodle-box poodle-modal-header__content c1"
  >
    ModalHeader
  </h2>
</div>
`;

exports[`<ModalHeader /> should render Box props correctly 1`] = `
.c0 {
  padding-top: 16px;
  padding-left: 24px;
  padding-right: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-style: normal;
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 133.33%;
  margin-top: 20px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 0;
  margin-bottom: 0;
}

.c2 {
  height: 2rem;
  width: 2rem;
  font-size: 2rem;
  margin-right: 8px;
  padding-left: 20px;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: 0;
  padding: 8px;
  pointer-events: initial;
  font-size: 1.5rem;
  color: #282828;
  cursor: pointer;
  background-color: transparent;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-transition: background-color 220ms ease-in,color 220ms ease-in;
  transition: background-color 220ms ease-in,color 220ms ease-in;
  border-radius: 50%;
  margin-left: auto;
  padding-left: 20px;
}

.c3:hover,
.c3:focus {
  color: #282828;
  background-color: #E0E0E0;
}

.c3:disabled {
  color: #A6A6A6;
  cursor: not-allowed;
}

<div
  class="poodle-box poodle-modal-header c0"
  id="id"
>
  <h2
    class="poodle-box poodle-modal-header__content c1"
  >
    <div
      class="poodle-box poodle-modal-header__icon c2"
    >
      <i>
        Icon
      </i>
    </div>
    ModalHeader
  </h2>
  <button
    aria-label="Close"
    class="poodle-box poodle-icon-button poodle-modal-header__close c3"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        clip-rule="evenodd"
        d="M8.071 6.343a1 1 0 1 0-1.414 1.414L10.899 12l-4.242 4.243a1 1 0 0 0 1.414 1.414l4.243-4.243 4.242 4.243a1 1 0 0 0 1.415-1.414L13.728 12l4.243-4.243a1 1 0 0 0-1.415-1.414l-4.242 4.243L8.07 6.343z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`<ModalHeader /> should render ModalHeader correctly 1`] = `
.c0 {
  padding-top: 16px;
  padding-left: 24px;
  padding-right: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-style: normal;
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 133.33%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 0;
  margin-bottom: 0;
}

<div
  class="poodle-box poodle-modal-header c0"
>
  <h2
    class="poodle-box poodle-modal-header__content c1"
  >
    ModalHeader
  </h2>
</div>
`;

exports[`<ModalHeader /> should render ModalHeader with auto id correctly 1`] = `
.c0 {
  padding-top: 16px;
  padding-left: 24px;
  padding-right: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-style: normal;
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 133.33%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 0;
  margin-bottom: 0;
}

<div
  class="poodle-box poodle-modal-header c0"
  id="id"
>
  <h2
    class="poodle-box poodle-modal-header__content c1"
  >
    ModalHeader
  </h2>
</div>
`;

exports[`<ModalHeader /> should render ModalHeader with close correctly 1`] = `
.c0 {
  padding-top: 16px;
  padding-left: 24px;
  padding-right: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-style: normal;
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 133.33%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 0;
  margin-bottom: 0;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: 0;
  padding: 8px;
  pointer-events: initial;
  font-size: 1.5rem;
  color: #282828;
  cursor: pointer;
  background-color: transparent;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-transition: background-color 220ms ease-in,color 220ms ease-in;
  transition: background-color 220ms ease-in,color 220ms ease-in;
  border-radius: 50%;
  margin-left: auto;
}

.c2:hover,
.c2:focus {
  color: #282828;
  background-color: #E0E0E0;
}

.c2:disabled {
  color: #A6A6A6;
  cursor: not-allowed;
}

<div
  class="poodle-box poodle-modal-header c0"
>
  <h2
    class="poodle-box poodle-modal-header__content c1"
  >
    ModalHeader
  </h2>
  <button
    aria-label="Close"
    class="poodle-box poodle-icon-button poodle-modal-header__close c2"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        clip-rule="evenodd"
        d="M8.071 6.343a1 1 0 1 0-1.414 1.414L10.899 12l-4.242 4.243a1 1 0 0 0 1.414 1.414l4.243-4.243 4.242 4.243a1 1 0 0 0 1.415-1.414L13.728 12l4.243-4.243a1 1 0 0 0-1.415-1.414l-4.242 4.243L8.07 6.343z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`<ModalHeader /> should render ModalHeader with icon correctly 1`] = `
.c0 {
  padding-top: 16px;
  padding-left: 24px;
  padding-right: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-style: normal;
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 133.33%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 0;
  margin-bottom: 0;
}

.c2 {
  height: 2rem;
  width: 2rem;
  font-size: 2rem;
  margin-right: 8px;
}

<div
  class="poodle-box poodle-modal-header c0"
>
  <h2
    class="poodle-box poodle-modal-header__content c1"
  >
    <div
      class="poodle-box poodle-modal-header__icon c2"
    >
      <i>
        Icon
      </i>
    </div>
    ModalHeader
  </h2>
</div>
`;

exports[`<ModalHeader /> theming should extend styles in global 1`] = `
.c0 {
  padding-top: 16px;
  padding-left: 24px;
  padding-right: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 10px;
  letter-spacing: 250px;
  top: 0;
  text-indent: 1px;
  font-style: normal;
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 133.33%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 0;
  margin-bottom: 0;
  margin-top: 13px;
  letter-spacing: 253px;
  top: 3px;
  text-indent: 4px;
}

.c2 {
  height: 2rem;
  width: 2rem;
  font-size: 2rem;
  margin-right: 8px;
  margin-top: 11px;
  letter-spacing: 251px;
  top: 1px;
  text-indent: 2px;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: 0;
  padding: 8px;
  pointer-events: initial;
  font-size: 1.5rem;
  color: #282828;
  cursor: pointer;
  background-color: transparent;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-transition: background-color 220ms ease-in,color 220ms ease-in;
  transition: background-color 220ms ease-in,color 220ms ease-in;
  border-radius: 50%;
  margin-left: auto;
  margin-top: 12px;
  letter-spacing: 252px;
  top: 2px;
  text-indent: 3px;
}

.c3:hover,
.c3:focus {
  color: #282828;
  background-color: #E0E0E0;
}

.c3:disabled {
  color: #A6A6A6;
  cursor: not-allowed;
}

<div
  class="poodle-box poodle-modal-header c0"
>
  <h2
    class="poodle-box poodle-modal-header__content c1"
  >
    <div
      class="poodle-box poodle-modal-header__icon c2"
    >
      <i>
        Icon
      </i>
    </div>
    ModalHeader
  </h2>
  <button
    aria-label="Close"
    class="poodle-box poodle-icon-button poodle-modal-header__close c3"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        clip-rule="evenodd"
        d="M8.071 6.343a1 1 0 1 0-1.414 1.414L10.899 12l-4.242 4.243a1 1 0 0 0 1.414 1.414l4.243-4.243 4.242 4.243a1 1 0 0 0 1.415-1.414L13.728 12l4.243-4.243a1 1 0 0 0-1.415-1.414l-4.242 4.243L8.07 6.343z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`<ModalHeader /> theming should extend styles in local 1`] = `
.c0 {
  padding-top: 16px;
  padding-left: 24px;
  padding-right: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 10px;
  letter-spacing: 250px;
  top: 0;
  text-indent: 1px;
  font-style: normal;
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 133.33%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 0;
  margin-bottom: 0;
  margin-top: 13px;
  letter-spacing: 253px;
  top: 3px;
  text-indent: 4px;
}

.c2 {
  height: 2rem;
  width: 2rem;
  font-size: 2rem;
  margin-right: 8px;
  margin-top: 11px;
  letter-spacing: 251px;
  top: 1px;
  text-indent: 2px;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: 0;
  padding: 8px;
  pointer-events: initial;
  font-size: 1.5rem;
  color: #282828;
  cursor: pointer;
  background-color: transparent;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-transition: background-color 220ms ease-in,color 220ms ease-in;
  transition: background-color 220ms ease-in,color 220ms ease-in;
  border-radius: 50%;
  margin-left: auto;
  margin-top: 12px;
  letter-spacing: 252px;
  top: 2px;
  text-indent: 3px;
}

.c3:hover,
.c3:focus {
  color: #282828;
  background-color: #E0E0E0;
}

.c3:disabled {
  color: #A6A6A6;
  cursor: not-allowed;
}

<div
  class="poodle-box poodle-modal-header c0"
>
  <h2
    class="poodle-box poodle-modal-header__content c1"
  >
    <div
      class="poodle-box poodle-modal-header__icon c2"
    >
      <i>
        Icon
      </i>
    </div>
    ModalHeader
  </h2>
  <button
    aria-label="Close"
    class="poodle-box poodle-icon-button poodle-modal-header__close c3"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        clip-rule="evenodd"
        d="M8.071 6.343a1 1 0 1 0-1.414 1.414L10.899 12l-4.242 4.243a1 1 0 0 0 1.414 1.414l4.243-4.243 4.242 4.243a1 1 0 0 0 1.415-1.414L13.728 12l4.243-4.243a1 1 0 0 0-1.415-1.414l-4.242 4.243L8.07 6.343z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`<ModalHeader /> theming should override styles 1`] = `
.c0 {
  margin-top: 50px;
  font-style: normal;
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 133.33%;
}

.c1 {
  margin-top: 53px;
}

.c2 {
  margin-top: 51px;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: 0;
  padding: 8px;
  pointer-events: initial;
  font-size: 1.5rem;
  color: #282828;
  cursor: pointer;
  background-color: transparent;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-transition: background-color 220ms ease-in,color 220ms ease-in;
  transition: background-color 220ms ease-in,color 220ms ease-in;
  border-radius: 50%;
  margin-top: 52px;
}

.c3:hover,
.c3:focus {
  color: #282828;
  background-color: #E0E0E0;
}

.c3:disabled {
  color: #A6A6A6;
  cursor: not-allowed;
}

<div
  class="poodle-box poodle-modal-header c0"
>
  <h2
    class="poodle-box poodle-modal-header__content c1"
  >
    <div
      class="poodle-box poodle-modal-header__icon c2"
    >
      <i>
        Icon
      </i>
    </div>
    ModalHeader
  </h2>
  <button
    aria-label="Close"
    class="poodle-box poodle-icon-button poodle-modal-header__close c3"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        clip-rule="evenodd"
        d="M8.071 6.343a1 1 0 1 0-1.414 1.414L10.899 12l-4.242 4.243a1 1 0 0 0 1.414 1.414l4.243-4.243 4.242 4.243a1 1 0 0 0 1.415-1.414L13.728 12l4.243-4.243a1 1 0 0 0-1.415-1.414l-4.242 4.243L8.07 6.343z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`<ModalHeader /> theming should render with default props correctly 1`] = `
.c0 {
  padding-top: 16px;
  padding-left: 24px;
  padding-right: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-style: normal;
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 133.33%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 0;
  margin-bottom: 0;
}

.c2 {
  height: 2rem;
  width: 2rem;
  font-size: 2rem;
  margin-right: 8px;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: 0;
  padding: 8px;
  pointer-events: initial;
  font-size: 1.5rem;
  color: #282828;
  cursor: pointer;
  background-color: transparent;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-transition: background-color 220ms ease-in,color 220ms ease-in;
  transition: background-color 220ms ease-in,color 220ms ease-in;
  border-radius: 50%;
  margin-left: auto;
}

.c3:hover,
.c3:focus {
  color: #282828;
  background-color: #E0E0E0;
}

.c3:disabled {
  color: #A6A6A6;
  cursor: not-allowed;
}

<div
  class="poodle-box poodle-modal-header hello c0"
  id="id"
>
  <h2
    class="poodle-box poodle-modal-header__content c1"
  >
    <div
      class="poodle-box poodle-modal-header__icon c2"
    >
      <i>
        Icon
      </i>
    </div>
    ModalHeader
  </h2>
  <button
    aria-label="Close"
    class="poodle-box poodle-icon-button poodle-modal-header__close c3"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        clip-rule="evenodd"
        d="M8.071 6.343a1 1 0 1 0-1.414 1.414L10.899 12l-4.242 4.243a1 1 0 0 0 1.414 1.414l4.243-4.243 4.242 4.243a1 1 0 0 0 1.415-1.414L13.728 12l4.243-4.243a1 1 0 0 0-1.415-1.414l-4.242 4.243L8.07 6.343z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;
